pandas/core/arrays/base.py

```diff
@@ -1452,6 +1448,31 @@ def _reduce(self, name: str, *, skipna: bool = True, **kwargs):
     # Non-Optimized Default Methods; in the case of the private methods here,
     #  these are not guaranteed to be stable across pandas versions.
 
+    def _hash_pandas_object(
```

Comment:
Something else: you added this method in the "Non-Optimized Default Methods" section, which says that private methods in this section are not guaranteed to be stable (in contrast to well established methods with an underscore that are meant to be overridden like `_values_for_factorize`, `_from_sequence`, etc, which are defined more above). 

Is that indeed the intention? (in that case it should not be mentioned in the documentation that this is a method to override? Currently it is added to the docstring higher up in this file) 
(If we don't want that external EAs override this, that's a another reason we should certainly keep using `_values_for_factorize`)

Reply:
> in contrast to well established methods with an underscore

Is that what underscores are supposed to mean?  I'm not eager to revive the EA-namespace-naming-scheme discussion, but that would fit well there.

IIRC the main motivation was allowing for a performant implementation for ArrowExtensionArray (without special-casing internally ArrowEA internally)

Reply:
> I am not sure which discussion needs to be revived (unless you don't agree that some of the methods with an underscore can be overridden by external EAs?)

I'm specifically _not_ interested in reviving the naming convention discussion.

I assume anything in the EA namespace is liable to be overridden by subclass authors.

Reply:
> I assume anything in the EA namespace is liable to be overridden by subclass authors.

The reason of my original comment here is because we have the following comment in the base EA class (one that you added I think):

https://github.com/pandas-dev/pandas/blob/d06f2d3dc412bd138778fdd26a29226efd0b3026/pandas/core/arrays/base.py#L1447-L1450

So while you say that you think anything in the EA namespace can be overridden by EA authors, the above comment at least seems to indicate there are some methods for which we guarantee stability, while for others it's at your own risk (to potentially keep it compatible with every new minor version). 

So essentially all that my original comment wanted to say is: if we want that EA authors need to implement `_hash_object_array` (as you also suggest in https://github.com/pandas-dev/pandas/issues/53501), we should move that method a bit higher up in this file to be above that comment about private methods not being stable.